x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/AsyncStatusResponse.java

```diff
@@ -145,9 +150,13 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVInt(successfulShards);
         out.writeVInt(skippedShards);
         out.writeVInt(failedShards);
-        if (isRunning == false) {
+        if (isRunning == false) { /// MP: Hmm, is this going to be a problem? Why is this not writeOptionalString?
             RestStatus.writeTo(out, completionStatus);
         }
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_008)) {
+            // optional since only CCS uses is; it is null for local-only searches
+            out.writeOptionalWriteable(clusters);
```

Comment:
Once we add new fields to Clusters, we will need to handle bwc within Clusters and all the classes that include a Clusters instance and serialize it will inherit that in their serialization logic? What would VersionedNamedWriteable help with in this case?

Reply:
OK, sounds good. Thanks for confirming this is the right approach then.